pkg/ingester/ingester.go

```diff
@@ -1601,12 +1601,11 @@ func (i *Ingester) pushSamplesToAppender(userID string, timeseries []mimirpb.Pre
 
 func (i *Ingester) QueryExemplars(ctx context.Context, req *client.ExemplarQueryRequest) (resp *client.ExemplarQueryResponse, err error) {
 	defer func() { err = i.mapReadErrorToErrorWithStatus(err) }()
-	if err := i.checkAvailableForRead(); err != nil {
-		return nil, err
-	}
-	if err := i.checkReadOverloaded(); err != nil {
+	finishReadRequestFn, err := i.startReadRequest()
```

Comment:
Nit: the `Fn` suffix is unnecessary, the type system already describes what type is each thing. Just like the function where we're in, `QueryExemplars` isn't `QueryExemplarsFn`.

(We don't do `errErr` or `labelNamesStr`, why would we do this to functions or channels?)